decred_wallet/Features/Wallet Setup/RecoverExistingWalletViewController.swift

```diff
@@ -102,65 +110,59 @@ class RecoverExistingWalletViewController: WalletSetupBaseViewController, UITabl
     func seedWordEntered(for wordIndex: Int, seedWord: String, moveToNextField: Bool) {
         self.userEnteredSeedWords[wordIndex] = seedWord
         
+        if (self.invalidSeedLabel.isHidden == false) {
+            self.invalidSeedLabel.isHidden = true
+            // increase top spacing so that confirm button is centered in display
+            self.tableViewFooterTopSpacingConstraint.constant = 30
+            UIView.animate(withDuration: 0.5) {
+                self.tableViewFooter.layoutIfNeeded()
+            }
+        }
+        
         if wordIndex < 32 && moveToNextField {
             self.focusSeedWordInput(at: wordIndex + 1)
         } else {
             self.view.endEditing(true)
         }
         
         if self.validateSeed().valid {
-            self.activateConfirmButton()
+            self.btnConfirm.backgroundColor = UIColor.appColors.decredGreen
         } else {
-            self.deactivateConfirmButton()
+            self.btnConfirm.backgroundColor = UIColor.appColors.lighterGray
         }
     }
     
     func focusSeedWordInput(at tableRowIndex: Int) {
         let tableIndexPath = IndexPath(row: tableRowIndex, section: 0)
-        
         let nextSeedWordCell = self.tableView.cellForRow(at: tableIndexPath) as? RecoveryWalletSeedWordCell
         nextSeedWordCell?.seedWordAutoComplete.becomeFirstResponder()
-        
-        self.tableView.scrollToRow(at: tableIndexPath, at: .middle, animated: true)
     }
-    
-    func activateConfirmButton() {
-        self.btnConfirm.backgroundColor = UIColor.appColors.decredGreen
-        self.lblEnterAllSeeds.isHidden = true
-        
-        // increase top spacing since warning label is now hidden so as to position button in center
-        self.tableViewFooterTopSpacingConstraint.constant = 30
-        UIView.animate(withDuration: 0.5) {
-            self.tableViewFooter.layoutIfNeeded()
-        }
+
+    @IBAction func backButtonTap(_ sender: Any) {
+        self.navigationController?.popViewController(animated: true)
     }
     
-    func deactivateConfirmButton() {
-        self.btnConfirm.backgroundColor = UIColor.appColors.lightGray
-        self.lblEnterAllSeeds.isHidden = false
-        
+    @IBAction func onConfirm() {
         if self.userEnteredSeedWords.contains("") {
-            self.lblEnterAllSeeds.text = "Not all seeds are entered. Please, check input fields and enter all seeds."
-        } else {
-            self.lblEnterAllSeeds.text = "You entered an incorrect seed. Please check your words."
+            self.displaySeedError("Not all seeds are entered. Please, check input fields and enter all seeds.")
+        } else{
```

Comment:
This should not require multiple reviews to correct. Please add a whitespace before the opening brace.